src/CSVArray.php

```diff
@@ -0,0 +1,75 @@
+<?php declare(strict_types=1);
+
+namespace MLL\Utils;
+
+use Exception;
+use Illuminate\Support\Arr;
+
+class CSVArray
+{
+    /**
+     * TODO: fix parsing multiline-content in csv.
+     *
+     * @return array<int, array<string, mixed>>
+     */
+    public static function toArray(string $csv, string $delimiter = ';'): array
+    {
+        $result = [];
+
+        $lines = StringUtil::splitLines($csv);
+
+        // The first row of the CSV usually contains the column headers
+        // We remove it so it does not show up in the result
+        $firstLine = array_shift($lines);
+        if (is_null($firstLine)) {
+            throw new Exception('Missing column headers.');
+        }
+
+        /** @var array<int, string> $columnHeaders */
+        $columnHeaders = str_getcsv($firstLine, $delimiter);
+
+        foreach ($lines as $index => $line) {
+            if (! StringUtil::hasContent($line)) {
+                continue;
+            }
+
+            $entries = str_getcsv($line, $delimiter);
+            if (count($entries) !== count($columnHeaders)) {
+                throw new Exception("The number of columns in row $index does not match the headers in CSV: " . $firstLine);
+            }
+
+            foreach ($columnHeaders as $columnIndex => $columnName) {
+                $result[$index + 1][$columnName] = $entries[$columnIndex];
+            }
+        }
+
+        return $result;
+    }
+
+    /**
+     * @param array<int, array<string, mixed>> $data
+     */
+    public static function toCSV(array $data, string $delimiter = ';'): string
+    {
+        if ([] === $data) {
+            throw new Exception('Array is empty');
+        }
+
+        // Use the keys of the array as the headers of the CSV
+        $headerLine = Arr::first($data);
+        assert(is_array($headerLine));
```

Comment:
This could go wrong at runtime if the user calls this method wrong. Should be an actual exception.